Hakawai/Mentions/HKWMentionsCreationStateMachineV2.m

```diff
@@ -165,11 +168,13 @@ - (void)stringInserted:(NSString *)string isWhitespace:(BOOL)isWhitespace isNewl
                                           ? HKWMentionsCreationActionWhitespaceCharacterInserted
                                           : HKWMentionsCreationActionNormalCharacterInserted);
                 [self.stringBuffer appendString:string];
-                // Fire off the request and start the timer
-                [self.dataProvider queryUpdatedWithKeyString:[self.stringBuffer copy]
-                                                  searchType:self.searchType
-                                                isWhitespace:isWhitespace
-                                            controlCharacter:self.explicitSearchControlCharacter];
+                if (self.dataProvider) {
+                    // Fire off the request and start the timer
+                    [self.dataProvider queryUpdatedWithKeyString:[self.stringBuffer copy]
+                                                      searchType:self.searchType
+                                                    isWhitespace:isWhitespace
+                                                controlCharacter:self.explicitSearchControlCharacter];
+                }
```

Comment:
if we are using custom chooser view delegate, we will need to call this to ask consumer of the library to update the TYAH query: 
[customChooserViewDelegate didUpdateKeyString:keyString
                                           controlCharacter:character];
customChooserViewDelegate is owned by mentions plugin

Reply:
same for other places

Reply:
i think you can just do this directly in creation state machine here: 
[self.delegate asyncRetrieveEntitiesForKeyString:[string copy]
                                          searchType:type
                                    controlCharacter:character
                                          completion:^(NSArray *results, BOOL dedupe,__unused BOOL isComplete)